packages/harmony/src/components/button/FilterButton/FilterButton.tsx

```diff
@@ -162,9 +153,13 @@ export const FilterButton = forwardRef<HTMLButtonElement, FilterButtonProps>(
       }
     }, [selection, variant, setIsOpen, setSelection])
 
-    const handleOptionSelect = useCallback((option: FilterButtonOption) => {
-      setSelection(option)
-    }, [])
+    const handleOptionSelect = useCallback(
+      (option: FilterButtonOption) => {
```

Comment:
I know the original wasn't doing this, but I'm wondering if there is use in skipping the state update and callback in the case where the selection isn't actually changing? We can't guarantee that the `onSelect` handler passed in is safe to call when the value hasn't actually changed.

Reply:
That's interesting... why would it be unsafe to call if the value doesn't change?

Reply:
"Unsafe" is probably too strong of a word. Unexpected might be more appropriate. Speaking as a person who has written logic in the past based around when a change event fires and the assumption that I need to re-render, recalculate or otherwise do some work in response, it is sometimes a headache when you receive one (or many) change handlers for values that haven't actually changed.
It's probably not a big deal, just convention I like to follow for reusable components.